### PR TITLE
DCMAW-7820: Create Passcode Info Screen

### PR DESCRIPTION
**Pull request by alex-bradbury**
- refactor of LoginRoutes, now an object
- creation of WelcomeScreenViewModel to inject necessary items
- creation of passcode info screen
- Update versions and create hilt test activity
- add tests

image

image

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -5,10 +5,13 @@ plugins {
     id("org.jlleitschuh.gradle.ktlint")
     id("uk.gov.onelogin.jvm-toolchains")
     id("com.google.dagger.hilt.android")
-
+    id("io.gitlab.arturbosch.detekt")
```
Comment (flavius-mester) [inline app/build.gradle.kts:8]: I can see that the id-check project was using detekt, so I am assuming that's why you added it to this project as well.
But I was just curious if we decided to use detect/sonar going forward? or can we use both?

**Review by flavius-mester — APPROVED** [review]
Great stuff 🚀 

Added some comments and some questions mostly for clarification and to start a conversation on those topics.

**But there's nothing here that should hold you from merging the PR**

We can change course later (on the testing for ex) if we decide to do so

File: app/build.gradle.kts (hunk 1 above)
Comment (alex-bradbury) [reply app/build.gradle.kts:8]: Think we actually use both on Id check currently, so can discuss if that's something we want to do or not

File: app/src/debug/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest
+    xmlns:android="http://schemas.android.com/apk/res/android">
+    <application>
+        <activity
+            android:name=".HiltTestActivity"
+            android:exported="true">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
```
[the remark this answers is not in the document]
Comment (alex-bradbury) [reply app/src/debug/AndroidManifest.xml:10]: I believe so, to allow the instrumentation tests to fire properly

File: app/build.gradle.kts [hunk 3]
```diff
@@ -91,6 +94,22 @@ android {
...
+                events = setOf(
+                    org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED,
+                    org.gradle.api.tasks.testing.logging.TestLogEvent.PASSED,
+                    org.gradle.api.tasks.testing.logging.TestLogEvent.SKIPPED
+                )
+            }
+        }
+        unitTests {
+            isReturnDefaultValues = true
+            isIncludeAndroidResources = true
```
[the remark this answers is not in the document]
Comment (alex-bradbury) [reply app/build.gradle.kts:109]: Yep that's a fair point, I tend to agree, might need to have a rethink about either implementation or testing approach

**Pull request merged**